### DISCUSSION TITLE
Google Cloud Storage Working but the image uploaded does not have GCS url

### DISCUSSION THREAD

**Discussion by harshalone** [directus/directus#14411] [Archive]
I recently configured GCS cloud storage into my directus instance. 

When I go to my admin and upload a file I can see that file uploaded to the GCS bucket.

but when I right click on the image and open it in another tab, the image uploaded does not have GCS url it has a directus url and cached as a parameter at the end.

I need to get the GCS url on the image field how do I do that?

**Comment by rijkvanzanten** [comment]
Directus doesn't currently return any file URLs to the storage adapter. Please give an upvote to https://github.com/directus/directus/discussions/7612 if that's something you'd like to see 👍🏻

**Comment by macieljrBiz** [comment]
Could you please share how did you configured it?
I'm trying all day with no success!
As no message is displayed with any error at the logs (I'm using docker compose, copied below, sensitive data removed), I have no clue why it's not working.

I believe, once it's working, I'll see it at File Library module, right?

compose-sample.yaml:
----------------------------------
version: '3'
services:

  directus:
    container_name: directus
    image: directus/directus:latest
    ports:
      - 8055:8055
    volumes:
      # Put a GCP Key file on Directus' root
      - ./directus/gcp-key-sample.json:/directus/gcp-key.json
      #- ./directus:/directus
      # By default, uploads are stored in /directus/uploads
      # Always make sure your volumes matches the storage root when using
      # local driver
      - ./directus/uploads:/directus/uploads
      # Make sure to also mount the volume when using SQLite
      - ./directus/database:/directus/database
      # If you want to load extensions from the host
      - ./directus/extensions:/directus/extensions
    networks:
      - directus
    environment:
      KEY: '25..b1'
      SECRET: '61..63'
      ADMIN_EMAIL: 'admin@example.com'
      ADMIN_PASSWORD: 'd1r3ctu5'
      STORAGE_LOCATIONS: 'local,gcs'
      STORAGE_LOCAL_DRIVER: 'local'
      STORAGE_GCS_DRIVER: 'gcs'
      STORAGE_GCS_ROOT: './uploads'
      STORAGE_GCS_KEY_FILENAME: './gcp-key.json'
      STORAGE_GCS_BUCKET: 'my-files'
      # Make sure to set this in production
      # (see https://docs.directus.io/self-hosted/config-options#general)
      # PUBLIC_URL: 'https://directus.example.com'

networks:
  directus:

gcp-key-sample.json (I use this same file with GCP's python lib on another project)
-----------------------------------
{
  "type": "service_account",
  "project_id": "prj",
  "private_key_id": "4455...59e6",
  "private_key": "-----BEGIN PRIVATE KEY-----\nMIIEvQI...jxivTFo=\n-----END PRIVATE KEY-----\n",
  "client_email": "someemail@appspot.gserviceaccount.com",
  "client_id": "11..87",
  "auth_uri": "https://accounts.google.com/o/oauth2/auth",
  "token_uri": "https://oauth2.googleapis.com/token",
  "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
  "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/ollo-staging%40appspot.gserviceaccount.com"
}
